mesonbuild/minstall.py

```diff
@@ -709,6 +710,13 @@ def run_install_script(self, d: InstallData, destdir: str, fullprefix: str) -> N
 
     def install_targets(self, d: InstallData, dm: DirMaker, destdir: str, fullprefix: str) -> None:
         for t in d.targets:
+            # In AIX, we archive our shared libraries.  When we install any package in AIX we need to
+            # install the archive in which the shared library exists. The below code does the same.
+            # We change the .so files having lt_version or so_version to archive file install. 
+            if is_aix():
```

Comment:
Shouldn't you be checking the machine here as well?

Reply:
So this installer class does not have the attribute environment.. How should we handle this?

Reply:
@dcbaker any more thoughts on this?